### PR TITLE
feat: centralize the common config templates used by the plugins

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description of the change:**
- Keep all common templates for the config directory centralized in /operator-sdk/internal/plugins/configbase/config/ instead of we have the code duplicated. 

**Motivation for the change:**
- Improve the maintainability see that any change will be required to be done just in one place. (e.g. If we need to update/push a patch from upstream it will be easily done.)
- It makes easier we know what has been or not customized per each type

Also, note that in the future we could have: 
a) a base config plugin which would be used for Ansible/Helm/Hibridys and any new language type (plugin phase 2)
OR
b) this implementation centralized in an SDK Operator lib. See; https://github.com/operator-framework/enhancements/pull/44
OR
c) Indeed, we can remove the full directory of the common config base and use them directly from kb if we make them public there until at least we have the plugin phase 2 in place.

### CODE REVIEW

File: internal/plugins/ansible/v1/scaffolds/init.go [hunk 1]
```diff
@@ -90,16 +92,22 @@ func (s *initScaffolder) scaffold() error {
...
+		// The role is customized for Ansible
+		&ansiblerbac.ManagerRole{},
 
 		&prometheus.Kustomization{},
 		&prometheus.ServiceMonitor{},
 
-		&manager.Manager{Image: imageName},
+		// Ansible do not use the default Manager because of the
+		// the resources limit configuration them do not work with it.
+		// Todo: https://github.com/operator-framework/operator-sdk/issues/3807
```
Comment (camilamacedo86) [inline internal/plugins/ansible/v1/scaffolds/init.go:104]:
```suggestion
		// Todo: https://github.com/operator-framework/operator-sdk/issues/3573
```

**Comment by joelanford** [comment]
I'm pretty sure we made an explicit decision to duplicate and keep these separate, just like the upstream kubebuilder project made a decision to duplicate and keep separate templates between versions of the Go plugin.

If we attempt to share, we cause problems for ourselves later. We plan to split helm and ansible into separate repos. So we'd have to:
1. Make yet another new repo to store these shared templates. We can't use `operator-lib` because that repo is meant ONLY for operator projects to import.
2. By putting these shared templates in a separate repo, they would have to be exported APIs which means others (even against our explicit warnings) would likely import them, which we don't want.
3. Any downstream mirrors of the plugin repos (e.g. for OpenShift) would have to _also_ downstream this shared repo, adding to the overhead of downstreaming code.
4. There would be nothing to prevent version skew of these imported libraries between the plugin repos, so there's no real guarantee that centralizing the templates would result in atomic changes to all plugins.
5. If one of the centralized templates needs to change in a specific way for just one operator type, then that template needs to be removed from the centralized repo and added to every repo importing that template, with that change applied to just the one repo requiring it. This is A LOT of overhead.
6. If a change is made in the centralized repo that works for one operator type but breaks another, we won't find out at least until the broken operator type tries to upgrade. If many other changes have gone in, it may be hard to unwind and fix things.

My vote would be to close this and make it the official position of the SDK project that templates will be duplicated between plugin types for the above reasons.

**Comment by camilamacedo86** [comment]
Hi @joelanford, 

Thank you for your input. Following some comments. 

> I'm pretty sure we made an explicit decision to duplicate and keep these separate, just like the upstream kubebuilder project made a decision to duplicate and keep separate templates between versions of the Go plugin.

IMO the Kubebuilder motivations to have v2/v3 do not apply to this scenario at all. See that the Kubebuilder Motivation for we have `v2/plugin` and `v3/plugin` makes simplicity which code is used in v2 and v3, allow us deprecated and remove v2 safely since we will be sure that has no code used by v3/plugin in v2/pugin. 

Now, let's think in SDK: This code is copy and past from Kubebuilder and we need to keep Kubebuilder/Helm/Ansible aligned and this code needs to be equals since came from Kubebuilder. See that when in Kubebuilder we say that v2/plugin and v3/plugin can be completed differently here we have the opposite the requirement.

If we attempt to share, we cause problems for ourselves later. We plan to split helm and ansible into separate repos. So we'd have to:

- Make yet another new repo to store these shared templates. We can't use operator-lib because that repo is meant ONLY for operator projects to import.

No. IMO so far we ought to use these files from Kubebuilder as LIB. See: 

> Indeed, we can remove the full directory of the common config base and use them directly from kb if we make them public there until at least we have the plugin phase 2 in place.

**Comment by joelanford** [comment]
> No. IMO so far we ought to use these files from Kubebuilder as LIB. See that, we can remove the files centralized here and use them directly from kb if we make them public there until at least we have the plugin phase 2 in place.

KB has expressly stated on many occasions that the raw template contents will not be made public. This was actually one or our early asks. Due to the position of the Kubebuilder community on this topic, we went down the path of plugins that could encapsulate their logic.

One thing I brought up 6ish months ago with the KB community is the idea of having separate plugins that manage separate parts of the project tree. So I could imagine a `kustomize` plugin that manages the `config` directory and a `go` plugin that manages files related to Go source code and its compilation/image building.

But I think we're pretty far off from that and there would need to be lots of discussion around it. All of which would likely be either alongside or after the phase 2 plugins work.

**Comment by camilamacedo86** [comment]
Hi @joelanford, 

Since shows that it is a topic which will bring a lot of discussions, I will close this PR. Just have it is good enough to illustrate the scenario in the  [EP](https://github.com/operator-framework/enhancements/pull/44/files). Thank you for your input.

**Pull request closed without merging**